pkg/aws/client.go

```diff
@@ -611,13 +613,53 @@ func (c *awsClient) FilterVPCsPrivateSubnets(subnets []*ec2.Subnet) ([]*ec2.Subn
 		}
 	}
 
+	// Temporary filter until local-zone and wave-length zones are supported by HCP
+	if isHostedCp {
+		privateSubnets, err = c.FilterSubnetsWithStandardAvailabilityZones(privateSubnets)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	if len(privateSubnets) < 1 {
 		return nil, fmt.Errorf("Failed to find private subnets associated with VPC '%s'", *subnets[0].VpcId)
 	}
 
 	return privateSubnets, nil
 }
 
+func (c *awsClient) FilterSubnetsWithStandardAvailabilityZones(subnets []*ec2.Subnet) ([]*ec2.Subnet, error) {
+	filteredSubnets := []*ec2.Subnet{}
+	subnetAvailabilityZonesList := []*string{}
+	standardAvailabilityZonesMap := map[string]bool{}
+
+	// list of availability zones to pass as input
+	for _, subnet := range subnets {
+		subnetAvailabilityZonesList = append(subnetAvailabilityZonesList, subnet.AvailabilityZone)
+	}
+
+	describeAvailabilityZonesOutput, err := c.ec2Client.DescribeAvailabilityZones(
+		&ec2.DescribeAvailabilityZonesInput{ZoneNames: subnetAvailabilityZonesList})
```

Comment:
What happens in case there are duplicates in the zone names? I would assume AWS will ignore the duplicates.

Reply:
Thanks for pointing this out.
From my manual test, I duplicates seem to be ignored.